contrib/upgrade.sh

```diff
@@ -25,7 +25,7 @@
 function giteacmd {
   if [[ $sudocmd = "su" ]]; then
     # `-c` only accept one string as argument.
-    "$sudocmd" - "$giteauser" -c "$giteabin --config $giteaconf --work-path $giteahome $@"
+    "$sudocmd" - "$giteauser" -c "\"$giteabin\" --config \"$giteaconf\" --work-path \"$giteahome\" \"$@\""
```

Comment:
```suggestion
    "$sudocmd" - "$giteauser" -c "\"$giteabin\" --config \"$giteaconf\" --work-path \"$giteahome\" $@"
```

Would `"\"$@\""` do the work correctly for all cases, eg: `"the" "foo bar"` and `"the" "foo" "bar"`?

If not, I suggest to remove the quoting at the moment, leave it as a potential bug, instead of an immediate bug.

Reply:
```suggestion
    "$sudocmd" - "$giteauser" -c "$(printf "%q " "$giteabin" "--config" "$giteaconf" "--work-path" "$giteahome" "$@")"
```


https://stackoverflow.com/questions/10835933/how-can-i-preserve-quotes-in-printing-a-bash-scripts-arguments

It might need to be like this to make all arguments quoted correctly.